### ISSUE TITLE
Messages from single chat are empty

### ISSUE THREAD

**Issue by LivelyAlex** [fabianonline/telegram_backup#42]
I tried to create a backup, and in general everything looks good, but messages from one chat are written to the database as empty, without text and data.

select count(*) from messages where messages.message_type='empty_message'
10241

i created log file with full output of program, with parameters --debug --trace --anonymize, maybe this helps. What else can I do to understand the problem?
![image](https://cloud.githubusercontent.com/assets/3461960/23941502/17ab936a-097a-11e7-8539-6848908c0e0d.png)
[log.txt](https://github.com/fabianonline/telegram_backup/files/844001/log.txt)

**Comment by LivelyAlex** [comment]
Can this be because this chat is a supergroup?

**Comment by fabianonline** [comment]
TL;DR: Quite possibly.

Disclaimer: This is knowledge about Telegram's internals I acquired by observing. I do not know for sure if this is correct, because there are no recent docs of Telegram's internals.

There is a fundamental difference between dialogs and groups on the one side and supergroups and channels on the other side: Messages sent in dialogs and groups get stored for each of the individual participants. Channels or supergroups have their own storage which is accessible for everyone joining the channel / supergroup. You can see this difference when joining a group: In a normal group, you won't see any of the older messages (unless someone forwards them to you). In a channel of supergroup, you can just scroll back and see "everything".

An `empty_message` is a message that once existed in your account, but doesn't do so anymore. I think what happened is that that group was at some time converted to a supergroup. At that moment, your copies of the group's messages became unnecessary and were deleted from your account.

Why do you think all those messages are from one specific chat? As far as I know they could be from any chat - the EmptyMessage object I receive from Telegram contains just the ID, nothing more. No way for me to find out to which chat this message belonged.

**Comment by LivelyAlex** [comment]
> Why do you think all those messages are from one specific chat?
in my telegram account i have ~10 small chats and 1 big with ~90% of all messages, more than 100 000 messages. Perhaps this 10 000 empty messages not from this chat, ok.
But then is there any way to create a backup copy of the supergroup using your program?

**Comment by fabianonline** [comment]
> But then is there any way to create a backup copy of the supergroup using your program?

At the moment not. I'm planning on adding backup functionality for supergroups and channels, but time is short at the moment and other features have a higher priority for me.

Is it okay to close this ticket?

**Comment by LivelyAlex** [comment]
yes, thank you
